cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -324,6 +325,47 @@ private KafkaRebalanceStatus updateStatus(KafkaRebalance kafkaRebalance,
         return rebalanceOptionsBuilder;
     }
 
+    private static Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> updateProgressFields(Reconciliation reconciliation,
+                                                                                              String host,
+                                                                                              int cruiseControlPort,
+                                                                                              CruiseControlApi apiClient,
+                                                                                              KafkaRebalance kafkaRebalance,
+                                                                                              ConfigMapOperator configMapOperator,
+                                                                                              MapAndStatus<ConfigMap, KafkaRebalanceStatus> desiredStatusAndMap) {
+        String configMapNamespace = kafkaRebalance.getMetadata().getNamespace();
+        String configMapName = kafkaRebalance.getMetadata().getName();
+
+        return configMapOperator.getAsync(configMapNamespace, configMapName)
+                .compose(existingConfigMap -> {
+                    ConfigMap desiredConfigMap = desiredStatusAndMap.getLoadMap();
+                    KafkaRebalanceStatus desiredStatus = desiredStatusAndMap.getStatus();
+                    KafkaRebalanceState desiredState = KafkaRebalanceUtils.rebalanceState(desiredStatus);
+
+                    if (existingConfigMap == null && desiredConfigMap == null) {
+                        return Future.succeededFuture(desiredStatusAndMap);
```

Comment:
```suggestion
                        return Future.succeededFuture(null);
```

I know the result is the same, but by using the variable you have to look at it (in the `if` condition) to know you are just returning `null` ... while this way it's pretty clear.

Reply:
Even though the existingConfigMap and desiredConfigMap are null here, we still need the contents desiredStatus to be returned!

Reply:
Ah sorry @kyguy I missread ... desiredConfigMap (which is null) vs desiredStatusAndMap you are returning!